### DISCUSSION TITLE
Getting Incorrect Values for Energy Metrics from IOPMPowerSource

### DISCUSSION THREAD

**Discussion by shivank-gupta-01** [YueChen-C/py-ios-device#160] [Q&A] [answered]
- We are using `py-ios-device` to calculate various device metrics.
- When calculating energy metrics in a running app on an iOS device, we’re noticing incorrect values:
- Most values are zero or lower than expected when the app is in a non-interactive state (e.g., when a video plays without user interaction).
- The library `IOPMPowerSource` appears to only provide reliable values when there is active user interaction within the app.
- We’re questioning if this is the expected behavior for this library.

**Comment by YueChen-C** [comment, marked as answer]
This is the device's Diagnostics data, we don't seem to be able to control or change it
